### PR TITLE
Plot OC-pO2 graph for SCR dives

### PR DESCRIPTION
**Pull request by willemferguson**
This commit follows on @glance-'s last PR for facilitating scr pO2
display when there is external pO2 monitoring. This PR
allows plotting the OC-equivalent pO2 graph for PSCR
dives. This happens in both the cases where there is no external
O2-monitoring AND when there is external pO2 monitoring. The
o2setpoint variable in plot-info is not affected at all. The
calculations are only done for PSCR dives and are achieved as
follows:

1) Within plot-info create a pressure-t called OC_pO2 in
   profile.h and populate this variable with the open-circuit
   pO2 values in profile.c.
2) Create a new partialPressureGasItem ocpo2GasItem in
   profilewidget2.h and, in profilewidget2.cpp, initialise it
   to read the plot-info OC_pO2 values and enable its
   display by using the setVisible method. The
   diveplotdatamodel was also touched in order to achieve
   this.

Signed-off-by: Willem Ferguson <willemferguson@zoology.up.ac.za>

### Describe the pull request:

- [ ] Bug fix
- [ ] Functional change
- [x] New feature
- [ ] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### Mentions:

@atdotde 
@glance- 
@sfuchs79

### CODE REVIEW

File: core/profile.c [hunk 1]
```diff
@@ -506,6 +506,8 @@ struct plot_data *populate_plot_entries(struct dive *dive, struct divecomputer *
 	struct plot_data *plot_data;
 	struct event *ev = dc->events;
 	(void) dive;
+	struct gasmix *gasmix = NULL; // These 2 pointers are needed by get_o2()
+	struct event *ev1 = NULL;     //
```
Comment (neolit123) [inline core/profile.c]: no need for this empty comment after `NULL;`

**Review by neolit123 — COMMENTED** [review]
added a small comment.
other than that the change LGTM.

others should add comments perhaps:
@atdotde @sfuchs79

**Comment by willemferguson** [comment]
This PR was submitted more than a month ago (Mar 14). Since then there has been no comment. Does this mean the code can be merged?

**Comment by dirkhh** [comment]
I was hoping that @atdotde  or @sfuchs79  would comment / approve as well.
This is code that I rarely ever touch and never ever use - so it's a lot harder for me to decide if this is ready to merge...

**Comment by atdotde** [comment]
Sorry, I missed this.

From a coding perspective: This looks good to me.

Could you explain, in which situation this "OC equivalent" becomes relevant, why would one look at this line?

Further, why don't you directly base this onto your branch which makes the divemode time depenent (which is in my opinion close to merging)?

**Comment by willemferguson** [comment]
Your first question. When diving SCR rebreather, an important factor is the oxygen drop over the mouthpiece, i.e. the difference between OC-equivalent pO2 and actually-measured pO2 in the loop of the rebreather; i.e. the vertical difference between the red and the green graphs in my initial screenshot. The larger the pO2 drop, the more decompression is added to the end of the dive because less oxygen usually means more nitrogen. However, the smaller the pO2 drop, the less efficient the rebreather is working and the SAC rate increases. On active SCR this is controlled by setting the rate of continuous gas addition. On pSCR this is controlled by opening or closing the gas release valve on the counterlung. So the vertical distance between the two graphs should ideally not be too large or too small, or at least managed within the objectives of the dive. In the case of a sticky or free-flowing injector, the red and green graphs start to converge, i.e. the closer one moves to an OC situation with little re-circulation of gas. Therefore sufficient distance between the red and green graphs is also an indication of well-working equipment. As indicated before, it is a situation very analogous to the setpoint of CCR equipment which provides a reference of how efficiently the gas addition management system works.

Your second question. This code is entirely independent of the divemode-dependent code, since pSCR dive logs is already part of the software.  As you know, the divemode-dependent branch primarily deals with bailout, which is an entirely different issue. Anyway, that is how I rationalised this to myself a month ago.

**Comment by glance-** [comment]
There's no "oxygen drop over the mouthpiece". The drop in oxygen levels are due to the oxygen the diver have metabolized.

The rest of your explanation makes clear why this is a interesting metric to look at.

There are a couple of small things I could point out in the code. I'd prefere another name than OC_pO2, because its SCR-specific, maybee SCR_OC_pO2?

Also, the connection between showPo2Changed and ocpo2GasItem, Should it really be there? Won't that cause it to show up if you toggle the graph on a non SCR-dive when viewing it? Or, atleast it might in my eyes. Anyway it looks odd.

Also, coloring the graph PO2_ALERT feels like a bad thing. Better to choose another color. Sort of hard to see whats a warning and not when more things start to have the "ALERT"-color. (I also think the o2SetpointGasItem should have another "base-color" but thats a separate issue).

**Comment by willemferguson** [comment]
@glance, your points could be easily addressed. With respect to your fear that the OC-pO2 might be displayed inappropriately, my intention was to add a button in the preferences to enable display of the graph, exactly in the way the setpoint for CCR dives is enabled. Would that solve your reservation? If so, should I add it here or as a separate PR?

**Comment by glance-** [comment]
Just add it to this PR.

**Comment by willemferguson** [comment]
@glance, I changed the colour of the OC pO2 graph to a shade of orange that still contrasts reasonably well with red (PO2_ALERT ). One wants a colour that contrasts well with the green of the PO2 graph as well as with the red. This is the best compromise I could find. Once this PR has been merged, I will fix the setpoint issue you referred to above. There is actually also a bug in the code for  setpoint display that needs rectification.

**Review by glance- — APPROVED** [review]

**Comment by neolit123** [comment]
hi, i will merge this given Anton has approved it the change looks OK to me.
thanks.

**Pull request merged**